ps3.py

```
@@ -0,0 +1,425 @@
+    dictionary, depending on how your code is structured).
+
+    Updates the hand: uses up the letters in the given word
+    and returns the new hand, without those letters in it.
+
+    Has no side effects: does not modify hand.
+
+    word: string
+    hand: dictionary (string -> int)
+    returns: dictionary (string -> int)
+    """
+    new_hand = hand.copy()
+
+    for letter in word.lower():
+        if letter in new_hand:
+            new_hand[letter] -= 1
+
+    for letter in hand:
+        if new_hand[letter] == 0:
+            new_hand.pop(letter)
+
+    return new_hand
+
+
+def is_valid_word(word, hand, word_list):
+    """
+    Returns True if word is in the word_list and is entirely
+    composed of letters in the hand. Otherwise, returns False.
+    Does not mutate hand or word_list.
+
+    word: string
+    hand: dictionary (string -> int)
+    word_list: list of lowercase strings
+    returns: boolean
+    """
+    word = word.lower()
+
+    for vowel in VOWELS:
+        normal_word = word.replace(WILDCARD, vowel)
+
+        if normal_word not in word_list:
+            return False
+
+        for letter in set(word):
+            if letter not in hand or word.count(letter) > hand[letter]:
+                return False
+
+    return True
+
+
+def calculate_handlen(hand):
+    """
+    Returns the length (number of letters) in the current hand.
+
+    hand: dictionary (string-> int)
+    returns: integer
+    """
+    letter_amount = 0
+
+    for letter in hand:
+        letter_amount += hand[letter]
+
+    return letter_amount
+
+
+def ask_user(text, warning_text, func):
+    """
+    text: message for input
+    warning_text: warning message
+    func: validation conditions
+    returns: value of variable
+    """
+    while True:
+        try:
+            n = input(text)
+            if not func(n):
+                raise ValueError
+            break
+        except ValueError:
+            print(warning_text)
```

Please - raise errors only in unexpected situations. Try implement this function without errors and try-except statements